appcues/src/main/java/com/appcues/Appcues.kt

```diff
@@ -41,28 +41,41 @@ public fun Appcues(
     accountId: String,
     applicationId: String,
     config: (AppcuesConfig.() -> Unit)? = null,
-): Appcues = Bootstrap
+): Appcues {
     // This creates the Scope and initializes the Appcues instance within, then returns the Appcues instance
     // ready to go with the necessary dependency configuration in its scope.
-    .createScope(
-        context = context,
-        config = AppcuesConfig(accountId, applicationId).apply { config?.invoke(this) }
-    ).get<Appcues>().also {
-        try {
-            // this is necessary to ensure that not only when we get a new token but whenever we initialize the application appcues should know
-            // whats the latest available token
-            FirebaseMessaging.getInstance().token.addOnCompleteListener {
-                // ensures we have token set
-                if (it.isSuccessful) {
-                    // store token on static pushToken that will be used on the next session start
-                    Appcues.pushToken = it.result
+    val scope = Bootstrap
+        .createScope(
+            context = context,
+            config = AppcuesConfig(accountId, applicationId).apply { config?.invoke(this) }
+        )
+
+    val appcues = scope.get<Appcues>()
+    val storage = scope.get<Storage>()
+    val sessionMonitor = scope.get<SessionMonitor>()
+
+    try {
+        // this is necessary to ensure that not only when we get a new token but whenever we initialize the application appcues should know
+        // whats the latest available token
+        FirebaseMessaging.getInstance().token.addOnCompleteListener {
+            // ensures we have token set
+            if (it.isSuccessful) {
+                if (sessionMonitor.hasSession()) {
+                    // can call setPushToken directly if we are already in a session, so the device_updated event is tracked
+                    appcues.setPushToken(it.result)
+                } else {
+                    // store token on static pushToken that will be used on the next session start to track device props
+                    storage.pushToken = it.result
```

Comment:
This reference to storage causes [this test](https://github.com/appcues/appcues-android-sdk/blob/main/appcues/src/test/java/com/appcues/AppcuesTest.kt#L63) to fail, as it does not use the mocked storage when the new scope is created here. Need to think about this one. I don't think it is a _real_ issue, but breaks that test.